drivers/vmwarevsphere/create.go

```diff
@@ -321,6 +312,30 @@ func (d *Driver) createFromLibraryName() error {
 	if err != nil {
 		return err
 	}
+	fr := vcenter.FilterRequest{Target: vcenter.Target{
+                        ResourcePoolID: d.resourcepool.Reference().Value,
+                        HostID:         hostId,
+                        FolderID:       folder.Reference().Value,
+        },
+        }
+
+	m := vcenter.NewManager(libManager.Client)
+	r, err := m.FilterLibraryItem(d.getCtx(), item.ID, fr)
+	if err != nil {
+                return err
+	}
+	if len(d.networks) != len(r.Networks) {
+		return fmt.Errorf("Mismatch in number of networks in content library template %s and rancher template", d.CloneFrom)
+	}
+	netIndex := 0
+	var nets []vcenter.NetworkMapping
+	for _, n := range d.networks {
+		nets = append(nets, vcenter.NetworkMapping{
+			Key:   r.Networks[netIndex],
```

Comment:
minor nit: `r.Networks[netIndex++]`

Reply:
will result in a syntax error. 
```
syntax error: unexpected ++, expecting :
```
A Go thing :)

Reply:
whoops